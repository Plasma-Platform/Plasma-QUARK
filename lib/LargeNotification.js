'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _typography = require('ui-toolkit/lib/typography');

var _typography2 = _interopRequireDefault(_typography);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LargeNotification = function (_Component) {
  (0, _inherits3.default)(LargeNotification, _Component);

  function LargeNotification() {
    (0, _classCallCheck3.default)(this, LargeNotification);
    return (0, _possibleConstructorReturn3.default)(this, (LargeNotification.__proto__ || (0, _getPrototypeOf2.default)(LargeNotification)).apply(this, arguments));
  }

  (0, _createClass3.default)(LargeNotification, [{
    key: 'render',
    value: function render() {
      var classes = (0, _classnames3.default)((0, _defineProperty3.default)({
        'notification_large': true
      }, this.props.className, this.props.className));

      var Text = _typography2.default[this.props.typographyCode];

      return _react2.default.createElement(
        'div',
        { className: classes, style: { width: this.props.width } },
        _react2.default.createElement('div', { className: 'notification_large__icon' }),
        _react2.default.createElement(
          'div',
          { className: 'notification_large__text' },
          _react2.default.createElement(
            Text,
            { type: 'default' },
            this.props.text
          )
        )
      );
    }
  }]);
  return LargeNotification;
}(_react.Component);

LargeNotification.propTypes = {
  className: _react.PropTypes.string,
  text: _react.PropTypes.string.isRequired,
  typographyCode: _react.PropTypes.oneOf(['T1', 'T2', 'T3', 'T4', 'T5', 'T6', 'H3']).isRequired,
  width: _react.PropTypes.string
};
exports.default = LargeNotification;