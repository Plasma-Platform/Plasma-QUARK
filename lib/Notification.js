'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _buttons = require('./buttons');

var _buttons2 = _interopRequireDefault(_buttons);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _isEmptyObject = require('./utils/isEmptyObject');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Notification = function (_Component) {
  (0, _inherits3.default)(Notification, _Component);

  function Notification() {
    var _ref;

    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, Notification);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Notification.__proto__ || (0, _getPrototypeOf2.default)(Notification)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      top: 'initial',
      left: 'initial',
      right: 'initial',
      bottom: 'initial'
    }, _this.getContainerWidth = function () {
      if (_this.container) _this.setState({ width: (_this.container.offsetWidth || 0) + 2 });
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  (0, _createClass3.default)(Notification, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      this.getContainerWidth();
    }
  }, {
    key: 'setPosition',
    value: function setPosition(coords) {
      this.setState(coords);
    }
  }, {
    key: 'render',
    value: function render() {
      var _classnames,
          _this2 = this;

      var classes = (0, _classnames3.default)((_classnames = {
        'quark-notification': true
      }, (0, _defineProperty3.default)(_classnames, 'quark-notification--' + this.props.size, true), (0, _defineProperty3.default)(_classnames, 'quark-notification--' + this.props.status, true), (0, _defineProperty3.default)(_classnames, 'quark-notification--' + this.props.position, true), (0, _defineProperty3.default)(_classnames, 'quark-notification--' + this.props.position + '-arrow', true), (0, _defineProperty3.default)(_classnames, 'quark-animated-tooltip_open_' + this.props.position, true), (0, _defineProperty3.default)(_classnames, this.props.className, this.props.className), _classnames));

      var parameters = {
        top: this.state.top,
        left: this.state.left,
        right: this.state.right,
        bottom: this.state.bottom,
        maxWidth: this.props.maxWidth,
        width: this.state.width
      };

      var Button = !(0, _isEmptyObject.isEmptyObject)(this.props.button) ? _buttons2.default[this.props.button.code ? this.props.button.code : 'B2J'] : null;
      var buttonAttr = {
        onClick: this.props.button.onClick,
        type: this.props.button.type,
        className: this.props.button.className
      };
      if (this.props.button.id) {
        buttonAttr.id = this.props.button.id;
      }
      return _react2.default.createElement(
        'div',
        {
          ref: function ref(container) {
            return _this2.container = container;
          },
          className: classes,
          style: parameters
        },
        _react2.default.createElement(
          'div',
          { className: 'quark-notification__container' },
          _react2.default.createElement(
            'span',
            { className: 'quark-notification__text' },
            this.props.text
          ),
          this.props.size === 'large' ? _react2.default.createElement(
            'div',
            { className: 'quark-notification__closeBlock' },
            _react2.default.createElement('div', {
              className: 'quark-notification__closeBlock__closeArea',
              onClick: this.props.onHideNotification,
              onTouchEnd: this.props.onHideNotification,
              id: this.props.closeIconId
            })
          ) : null
        ),
        Button ? _react2.default.createElement(
          Button,
          buttonAttr,
          this.props.button.text
        ) : null
      );
    }
  }]);
  return Notification;
}(_react.Component);

Notification.propTypes = {
  className: _react.PropTypes.string,
  text: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.array, _react.PropTypes.object]).isRequired,
  button: _react.PropTypes.object,
  size: _react.PropTypes.string.isRequired,
  maxWidth: _react.PropTypes.string,
  closeIconId: _react.PropTypes.string,
  onHideNotification: _react.PropTypes.func,
  status: _react.PropTypes.oneOf(['default', 'success', 'error']).isRequired,
  position: _react.PropTypes.oneOf(['top', 'bottom', 'left', 'right']).isRequired
};
exports.default = Notification;