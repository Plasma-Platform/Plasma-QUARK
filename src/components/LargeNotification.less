@import '../assets/styles/palette.less';

@exclamation-mark : url('./LargeNotification/exclamation-mark.svg');

.notification_large {
    display: flex;
    align-items: center;
    padding: 20px;
    background-color: @lagre-hint-bg;

    max-width: 100%;
    box-sizing: border-box;

    user-select: none;

    &__icon {
        margin-right: 20px;

        width: 40px;
        height: 40px;
        background: @exclamation-mark no-repeat;
    }

    &__text {
        width: calc(~"100% - 60px");
        color: @large-hint-text;
    }
}
