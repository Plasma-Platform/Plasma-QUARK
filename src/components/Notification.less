@import '../assets/styles/mixins.less';
@import '../assets/styles/palette.less';

.quark-notification {
  position: absolute;
  border-radius: 3px;
  box-shadow: 0px 5px 10px 0px @hint-shadow-default;
  font-size: 14px;
  font-family: PT_Sans;
  color: #fff;
  z-index: 1000;

  //it's needed for eye-showing tooltip but not for others
  .password-toggle & {
    pointer-events: none;
  }

  &__text {
    width: 100%;
    z-index: 1000;
    white-space: nowrap;
  }

  &__closeBlock {
    cursor: pointer;
    pointer-events: auto;
    margin-left: 20px;
    align-self: flex-start;
    position: relative;
    width: 10px;
    height: 10px;
    background: url('./Notification/close-cross.svg') no-repeat;

    &__closeArea {
      cursor: pointer;
      position: absolute;
      width: 40px;
      height: 40px;

      left: -15px;
      top: -15px;
    }
  }

  &__container {
    padding: 0 20px;
    z-index: 1000;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 3px;
  }

  &--left-arrow {
    &:after {
      .common-pseudo-for-hint();
      box-shadow: 4px 0px 10px 0px;
      margin-top: -10px;
      top: 50%;
      right: -6px;
    }
  }

  &--right-arrow {
    &:before {
      box-shadow: -2px 2px 10px 0px;
      .common-pseudo-for-hint();
      margin-top: -10px;
      top: 50%;
      left: -6px;
    }
  }

  &--top-arrow {
    &:before {
      box-shadow: 3px 5px 10px 0px;
      .common-pseudo-for-hint();
      margin-left: -8px;
      bottom: -8px;
      left: 50%;
    }
  }

  &--bottom-arrow {
    &:after {
      box-shadow: -1px 1px 10px 0px;
      .common-pseudo-for-hint();
      margin-left: -8px;
      top: -8px;
      left: 50%;
    }
  }

  &--default {
    background-color: @hint-grey;
    &:before, &:after {
      background-color: @hint-grey;
      color: @hint-shadow-default;
    }
    .notification__container {
      background-color: @hint-grey;
    }
  }

  &--success {
    box-shadow: 0px 5px 10px 0px @hint-shadow-success;
    &:before, &:after {
      background-color: @hint-green;
      color: @hint-shadow-success;
    }
    .notification__container {
      background-color: @hint-green;
    }
  }

  &--error {
    box-shadow: 0px 5px 10px 0px @hint-shadow-error;
    &:before, &:after {
      background-color: @hint-red;
      color: @hint-shadow-error;
    }
    .notification__container {
      background-color: @hint-red;
    }
  }

  &--small {
    .notification__container {
      height: 40px;
    }
  }

  &--large {
    .notification__container {
      padding: 20px;
    }
    min-height: 20px;
    justify-content: space-between;
  }

  &--large &__text {
    width: calc(~"100% - 30px");
    white-space: normal;
  }
}
