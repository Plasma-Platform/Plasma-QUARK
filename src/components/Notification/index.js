export * from './types/';
export default from './Notification';
