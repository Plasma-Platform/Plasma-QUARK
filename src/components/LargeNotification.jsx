import React, {Component, PropTypes} from 'react';
import classnames from 'classnames';

import Typography from 'ui-toolkit/lib/typography';

import './LargeNotification.less';

export default class LargeNotification extends Component {
  static propTypes = {
    className      : PropTypes.string,
    text           : PropTypes.string.isRequired,
    typographyCode : PropTypes.oneOf([
      'T1', 'T2', 'T3', 'T4', 'T5', 'T6', 'H3']).isRequired,
    width: PropTypes.string
  }

  render () {
    const classes = classnames({
      'notification_large'   : true,
      [this.props.className] : this.props.className
    });

    const Text = Typography[this.props.typographyCode];

    return (
      <div className={classes} style={{width: this.props.width}}>
        <div className='notification_large__icon'/>

        <div className='notification_large__text'>
          <Text type='default'>{this.props.text}</Text>
        </div>
      </div>
    );
  }
}
