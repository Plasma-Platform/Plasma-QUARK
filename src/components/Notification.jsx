import React, {Component, PropTypes} from 'react';
import Buttons from './buttons';
import classnames                      from 'classnames';
import {isEmptyObject} from './utils/isEmptyObject';

import '../assets/styles/animations.less';
import './Notification.less';

export default class Notification extends Component {
  static propTypes = {
    className          : PropTypes.string,
    text               : PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.object]).isRequired,
    button             : PropTypes.object,
    size               : PropTypes.string.isRequired,
    maxWidth           : PropTypes.string,
    onHideNotification : PropTypes.func,
    status             : PropTypes.oneOf(['default', 'success', 'error']).isRequired,
    position           : PropTypes.oneOf(['top', 'bottom', 'left', 'right']).isRequired
  };

  state = {
    top    : 'initial',
    left   : 'initial',
    right  : 'initial',
    bottom : 'initial'
  };

  componentDidMount () {
    this.getContainerWidth();
  };

  getContainerWidth = () => {
    if (this.container) this.setState({width: this.container.offsetWidth + 2});
  };

  setPosition (coords) {
    this.setState(coords);
  };

  render () {
    const classes = classnames({
      'notification'                                   : true,
      [`notification--${this.props.size}`]             : true,
      [`notification--${this.props.status}`]           : true,
      [`notification--${this.props.position}`]         : true,
      [`notification--${this.props.position}-arrow`]   : true,
      [`animated-tooltip_open_${this.props.position}`] : true,
      [this.props.className]                           : this.props.className
    });

    const parameters = {
      top      : this.state.top,
      left     : this.state.left,
      right    : this.state.right,
      bottom   : this.state.bottom,
      maxWidth : this.props.maxWidth,
      width    : this.state.width
    };

    const Button = !isEmptyObject(this.props.button) ? Buttons[this.props.button.code ? this.props.button.code : 'B2J'] : null;
    return (
      <div
        ref={container => this.container = container}
        className={classes}
        style={parameters}
      >
        <div className='notification__container'>
          <span className='notification__text' dangerouslySetInnerHTML={{__html: this.props.text}}>{this.props.children}</span>
          {this.props.size === 'large'
            ? (<div className='notification__closeBlock'>
            <div
              className='notification__closeBlock__closeArea'
              onClick={this.props.onHideNotification}
              onTouchEnd={this.props.onHideNotification}
            />
          </div>)
            : null
          }
        </div>
        {Button
          ? (
            <Button action={this.props.button.action} type={this.props.button.type} className={this.props.button.className}>{this.props.button.text}</Button>
          )
          : null
        }
      </div>
    );
  }
}
