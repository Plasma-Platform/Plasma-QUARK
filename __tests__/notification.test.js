/* global expect:false, test:false, it:false, describe:false, jest:false, beforeEach:false */

import React from 'react';
import {shallow, mount} from 'enzyme';

import LargeNotification from '../src/components/LargeNotification.jsx';
import Notification from '../src/components/notifications';

describe('Notification view testing', () => {
  let notifications = [];

  beforeEach(() => {
    notifications = Object.keys(Notification);
  });

  it('takes a snapshot of Large Notification moved from UI-toolkit ', () => {
    let shot = shallow(<LargeNotification text="test" typographyCode="T1" />).html();
    expect(shot).toMatchSnapshot();
  });

  it('takes snapshots of all type of notifications', () => {
    let Component;
    let shot;
    notifications.forEach((type) => {
      Component = Notification[type];
      shot = shallow(<Component text="test" typographyCode="T1" onHideNotification={() => {}} />).html();
      expect(shot).toMatchSnapshot();
    });
  });

  it('sets position of notification', () => {
    let Component;
    let shot;
    const coords = {
      top  : 0,
      left : 0
    };

    notifications.forEach((type) => {
      Component = Notification[type];
      shot = mount(<Component text="test" typographyCode="T1" onHideNotification={() => {}} />);
      shot.node.setPosition = jest.fn(shot.node.setPosition);
      shot.node.setPosition(coords);
      expect(shot.node.setPosition).toBeCalled();
    });
  });
});
