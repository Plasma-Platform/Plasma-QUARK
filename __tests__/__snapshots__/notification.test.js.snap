exports[`Notification view testing takes a snapshot of Large Notification moved from UI-toolkit  1`] = `"<div class=\"notification_large\"><div class=\"notification_large__icon\"></div><div class=\"notification_large__text\"><span class=\"TMUI__TypographyText TMUI__TypographyText--light TMUI__TypographyText--default TMUI__TypographyText--1\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 1`] = `"<div class=\"notification notification--small notification--default notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 2`] = `"<div class=\"notification notification--small notification--default notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 3`] = `"<div class=\"notification notification--small notification--default notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 4`] = `"<div class=\"notification notification--small notification--default notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 5`] = `"<div class=\"notification notification--large notification--default notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 6`] = `"<div class=\"notification notification--large notification--default notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 7`] = `"<div class=\"notification notification--large notification--default notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 8`] = `"<div class=\"notification notification--large notification--default notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 9`] = `"<div class=\"notification notification--small notification--error notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 10`] = `"<div class=\"notification notification--small notification--error notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 11`] = `"<div class=\"notification notification--small notification--error notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 12`] = `"<div class=\"notification notification--small notification--error notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 13`] = `"<div class=\"notification notification--large notification--error notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 14`] = `"<div class=\"notification notification--large notification--error notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 15`] = `"<div class=\"notification notification--large notification--error notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 16`] = `"<div class=\"notification notification--large notification--error notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 17`] = `"<div class=\"notification notification--small notification--success notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 18`] = `"<div class=\"notification notification--small notification--success notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 19`] = `"<div class=\"notification notification--small notification--success notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 20`] = `"<div class=\"notification notification--small notification--success notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 21`] = `"<div class=\"notification notification--large notification--success notification--top notification--top-arrow animated-tooltip_open_top\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 22`] = `"<div class=\"notification notification--large notification--success notification--bottom notification--bottom-arrow animated-tooltip_open_bottom\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 23`] = `"<div class=\"notification notification--large notification--success notification--right notification--right-arrow animated-tooltip_open_right\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 24`] = `"<div class=\"notification notification--large notification--success notification--left notification--left-arrow animated-tooltip_open_left\" style=\"top:initial;left:initial;right:initial;bottom:initial;\"><div class=\"notification__container\"><span class=\"notification__text\">test</span><div class=\"notification__closeBlock\"><div class=\"notification__closeBlock__closeArea\"></div></div></div></div>"`;

exports[`Notification view testing takes snapshots of all type of notifications 25`] = `"<div class=\"notification_large\"><div class=\"notification_large__icon\"></div><div class=\"notification_large__text\"><span class=\"TMUI__TypographyText TMUI__TypographyText--light TMUI__TypographyText--default TMUI__TypographyText--1\">test</span></div></div>"`;
